src/module_2/module_2.md

```diff
@@ -0,0 +1,3512 @@
+    1123           NaN             NaN           NaN         NaN  
+    1958           NaN             NaN           NaN         NaN  
+    675            NaN             NaN           NaN         NaN  
+    4694           NaN             NaN           NaN         NaN  
+    -- The df named df_users has been saved
+    
+
+
+
+```python
+# We have saved the 5 dfs inside a dict called dfs
+# We have saved each df with its name df_{parquet_file}
+```
+
+# EDA (in progress)
+
+We have been presented with different datasets from a groceries e-commerce platform selling directly to consumers. In that sense, we have 5 different tables with different info in each of them.
+
+- df_orders: An order history of customers. Each row is an order and the
+item_ids for the order are stored as a list in the item_ids column
+- df_regulars:  Users are allowed to specify items that they wish to buy
+regularly. This data gives the items each user has asked to get regularly, along
+with when they input that information.
+- df_abandoned cart: If a user has added items to their basket but not
+bought them, we capture that information. Items that were abandoned are stored
+as a list in item_ids.
+- df_inventory: Some information about each item_id
+- df_users: Information about users.
+
+1. Priorly to working with data, we will try to inspect each of the dataframes in order to understand the information. With the description of each table provided, we would try to figure out how the data looks like - then we could start thinking of some hypothesis and proceed with its verification.
+
+2. We may display them, inspect columns and dtypes, verify if everythings makes sense about the metadata. After that, we could try to look for possible NANs or Null values, doublechecking data quality is OK. If we find any issue or problem during the process, with the prior knowledge obtained, we should be able to make a decision and justify the solution applied.
+
+3. Once data has been fixed, we may join or transform them in order to work with that in future steps.
+
+3. After all, we will formulate the corresponding hypothesis tested about the data plus some commentaries and additional info we have found.
+
+### Orders
+
+An orders history of customers. Each row is an order and the
+item_ids for the order are stored as a list in the item_ids column
+
+
+
+```python
+print(df_orders.shape)
+df_orders.head()
+```
+
+    (8773, 6)
+
+
+
+
+
+<div>
+<style scoped>
+    .dataframe tbody tr th:only-of-type {
+        vertical-align: middle;
+    }
+
+    .dataframe tbody tr th {
+        vertical-align: top;
+    }
+
+    .dataframe thead th {
+        text-align: right;
+    }
+</style>
+<table border="1" class="dataframe">
+  <thead>
+    <tr style="text-align: right;">
+      <th></th>
+      <th>id</th>
+      <th>user_id</th>
+      <th>created_at</th>
+      <th>order_date</th>
+      <th>user_order_seq</th>
+      <th>ordered_items</th>
```

Comment:
it's a bit weird that we use the word `items` while in the other tables are `variants`